New customer-facing Academy page with hero section, audience focus, and CTAs

Vincamine opened the pull request:
**Description**

This PR fixes ##6784

**Notes for Reviewers**
This page divied into 2 parts:
1. Hero section: ensures customers immediately understand what we offer.
2. Audience – Learner-Facing and Enterprise-Facing: The content is tailored so each audience sees relevant benefits. I've added Links and Calls to Action (CTAs): Clear pathways (links, CTAs) guide users on what they can explore next.

(also fixed the Gatsby Link error)

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.
 

<!--
Thank you for contributing to Layer5 projects! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
5. Build and test your changes before submitting a PR. 
6. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

l5io commented:
🚀 Preview for commit 641a17ca52e8b65904b8bf36ba528d1ea943f325 at: https://68be1596d1a80a11b54230b7--layer5.netlify.app

leecalcote commented on src/components/Academy/Enterprise-Section/EnterpriseSection.js:
```diff
@@ -0,0 +1,283 @@
+import styled from "styled-components";
+
+const EnterpriseSectionWrapper = styled.section`
+  font-family: "Qanelas Soft", sans-serif;
```
Why is this not standard? @vr-varad @ARYANSHAH1567, let’s be on a mission to ensure that we are reusing components and reusing styles across this site without defining new styles. Please make this a topic of discussion for tomorrow’s meeting.

Vincamine commented:
- Hero section image: Lightcone and Academy 
- Refactored Hero section configuration based on Canvas settings to ensure consistency
- Link & CTA standardization: standardized all links and CTA buttons for consistent behavior and styling.
- Hero section overall refinement
- Removed unused pseudo-element
- Standardized text formatting

l5io commented:
🚀 Preview for commit 4bda85a53b0eeba3427e490e57197163a79eef2a at: https://68bfbf9e34db550d30f917d5--layer5.netlify.app

Namanv0509 commented:
<img width="1296" height="770" alt="image" src="https://github.com/user-attachments/assets/a63517dc-035e-4964-949c-d35982b19d97" />
These cards gets elevated and cursor changes to pointer on hover , which is not desired if they are not hyperlinked .

leecalcote commented on src/components/Academy/Enterprise-Section/EnterpriseSection.js:
```diff
@@ -0,0 +1,269 @@
+            <Button
+              $primary
+              title="View Pricing"
+              $external={true}
```
This is an internal URL.

leecalcote commented on src/components/Academy/Enterprise-Section/EnterpriseSection.js:
```diff
@@ -0,0 +1,269 @@
+              $primary
+              title="View Pricing"
+              $external={true}
+              $url="https://layer5.io/pricing"
```
Please change this to be a relative path. `/pricing`.
